backend.ps1

```diff
@@ -69,19 +70,27 @@ class AzureInstance : Instance {
 }
 
 class HypervInstance : Instance {
-    [String] $VHDPath
+    [String] $BootVHDPath
     [String] $DvdDrive
 
     HypervInstance ($Backend, $Name) : base ($Backend, $Name) {
     }
 
-    [void] CreateInstance () {
-        $this.Backend.CreateInstance($this.Name, $this.VHDPath)
+    [void] Create () {
+        $this.BootVHDPath = $this.Backend.CreateInstance($this.Name, $this.BootVHDPath)
```

Comment:
why do you need this, is about the suffix? then do not send "false" or erroneous params to the backend just to overwrite them beforehand.

Reply:
in case we already have the instance object and want to use the vhdpath stored in it

in backend i check if the param is null or empty and if so, i make it on the spot